src/theory/builtin/theory_builtin_rewriter.cpp

```diff
@@ -100,10 +100,26 @@ RewriteResponse TheoryBuiltinRewriter::postRewrite(TNode node) {
       {
         if (node[1][i] == node[0][0])
```

Comment:
Please add the high-level abstraction of the rewrite, as is in the PR description